index.js

```diff
@@ -391,6 +392,18 @@ function Dyno(options) {
     delete dynoExtensions.queryStream;
     delete dynoExtensions.scanStream;
   }
+  
+  Object.entries(nativeFunctions).forEach(function ([k, v]) {
+    if (typeof v === 'function') {
+      nativeFunctions[`${k}Async`] = promisify(v);
+    }
+  });
+
+  ['createTable', 'deleteTable', 'query', 'scan'].forEach(function(method) {
+    if (typeof dynoExtensions[method] === 'function') {
+      dynoExtensions[`${method}Async`] = promisify(dynoExtensions[method]);
+    }
+  });
```

Comment:
All of these are functions, and we are creating these objects ourselves, it's unlikely that there will be any unexpected values. Do we really the additional verification?

I'd prefer if we kept this code as simple as possible and maybe add a test that all of these things have the expected type, so that if this complexity is unavoidable, it's handled in CI, not in production code.

Also I think keeping the two loops as similar as possible makes them easier to understand, what would you say about this version?

```suggestion
  for(const name of Object.keys(nativeFunctions) {
    nativeFunctions[`${name}Async`] = promisify(nativeFunctions[name]);
  }

  for(const name of ['createTable', 'deleteTable', 'query', 'scan']) {
    dynoExtensions[`${name}Async`] = promisify(dynoExtensions[name]);
  }
```

Reply:
The check for `['createTable', 'deleteTable', 'query', 'scan']` is necessary as some of them might not exist in write only or read only mode. But of course they can be easily checked by `if (dynoExtensions[name])`